src/libs/workspace-utils.ts

```diff
@@ -108,6 +109,34 @@ export const containsProtectedDataPolicy = (policies: WorkspacePolicy[] | undefi
 export const protectedDataMessage =
   'Enhanced logging and monitoring are enabled to support the use of protected or sensitive data in this workspace.';
 
+export const hasRegionConstraint = (workspace: BaseWorkspace): boolean =>
+  getRegionConstraintLabels(workspace.policies).length > 0;
+
+export const getRegionConstraintLabels = (policies: WorkspacePolicy[] | undefined): string[] => {
+  const regionPolicies = _.filter(
+    (policy) => policy.namespace === 'terra' && policy.name === 'region-constraint',
+    policies
+  );
+  const regionLabels: string[] = [];
+  _.forEach((policy) => {
+    _.forEach((data) => {
+      if ('region-name' in data) {
+        const region = data['region-name'];
+        const regionName = region.startsWith('azure.') ? region.split('azure.')[1] : region;
+        regionLabels.push(_.has(regionName, azureRegions) ? azureRegions[regionName].label : regionName);
```

Comment:
I decided that if we can't parse the region into something we recognize, it's better to show them what the policy actually specifies vs. a generic "unknown" value.